File: src/web/src/stores/game.js
import { defineStore } from 'pinia'
import axios from 'axios'
import { useRouter } from 'vue-router'

const API_BASE = '/api/game'

export const useGameStore = defineStore('game', {
  state: () => ({
    gameId: null,
    state: null,
    lastEvent: null,     // 最近的事件结果
    pendingChoice: null, // 等待玩家选择（祝福/奇物列表）
    loading: false,
    error: null,
  }),
  
  actions: {
    async startGame() {
      this.loading = true
      try {
        const res = await axios.post(`${API_BASE}/start`)
        this.gameId = res.data.game_id
        this.state = res.data.state
        return res.data.game_id
      } catch (e) {
        this.error = e.message
      } finally {
        this.loading = false
      }
    },
    
    async fetchState() {
      if (!this.gameId) return
      try {
        const res = await axios.get(`${API_BASE}/${this.gameId}`)
        this.state = res.data
      } catch (e) {
        this.error = e.message
      }
    },
    
    async playCard(index) {
      this.loading = true
      try {
        const res = await axios.post(`${API_BASE}/${this.gameId}/play`, {
          card_index: index
        })
        this.lastEvent = res.data
        // 检查是否需要选择祝福/奇物
        if (res.data.options) {
          this.pendingChoice = res.data.options
        } else {
          this.pendingChoice = null
        }
        await this.fetchState()
        return res.data
      } catch (e) {
        this.error = e.message
      } finally {
        this.loading = false
      }
    },
    
    async chooseBlessing(index) {
      try {
        const res = await axios.post(`${API_BASE}/${this.gameId}/choose-blessing`, null, {
          params: { index }
        })
        this.pendingChoice = null
        await this.fetchState()
        return res.data
      } catch (e) {
        this.error = e.message
      }
    },
    
    async chooseCurio(index) {
      try {
        const res = await axios.post(`${API_BASE}/${this.gameId}/choose-curio`, null, {
          params: { index }
        })
        this.pendingChoice = null
        await this.fetchState()
        return res.data
      } catch (e) {
        this.error = e.message
      }
    },
  },
})

File: src/api/routers/game.py
from fastapi import APIRouter, HTTPException
from ..models import GameState, PlayCardRequest, PlayCardResponse
from ..game_manager import game_manager

router = APIRouter(prefix="/api/game", tags=["game"])

@router.post("/start")
def start_game():
    game_id = game_manager.create_game()
    state = game_manager.get_state(game_id)
    return {"game_id": game_id, "state": state}

@router.get("/{game_id}")
def get_state(game_id: str):
    state = game_manager.get_state(game_id)
    if not state:
        raise HTTPException(status_code=404, detail="Game not found")
    return state

@router.post("/{game_id}/play")
def play_card(game_id: str, req: PlayCardRequest):
    result = game_manager.play_card(game_id, req.card_index)
    if not result:
        raise HTTPException(status_code=404, detail="Game not found")
    return result

@router.post("/{game_id}/choose-blessing")
def choose_blessing(game_id: str, index: int):
    game = game_manager.get_game(game_id)
    if not game:
        raise HTTPException(status_code=404, detail="Game not found")
    result = game["run"].choose_blessing(index)
    return result

@router.post("/{game_id}/choose-curio")
def choose_curio(game_id: str, index: int):
    game = game_manager.get_game(game_id)
    if not game:
        raise HTTPException(status_code=404, detail="Game not found")
    result = game["run"].choose_curio(index)
    return result